File: spec/javascripts/sort/MergeSortSpec.js
describe('MergeSort', function() {

  describe('sort', function() {
    it('should sort a collection', function() {
      var ms = new MergeSort();

      expect(ms.sort([9,8,7,6,5,4,3,2,1,0])).toEqual([0,1,2,3,4,5,6,7,8,9]);
      expect(ms.sort([5,4,3,2,1])).toEqual([1,2,3,4,5]);
      expect(ms.sort([78,34,2,5,87,0])).toEqual([0,2,5,34,78,87]);
    });
  });
});

File: spec/lib/sort/Merge_Sort_Spec.rb
require 'spec_helper'

describe 'Merge Sort' do

  describe 'sort' do
    it 'should sort a collection' do
      ms = Merge_Sort.new
      collection = [9,8,7,6,5,4,3,2,1,0]
      sorted_collection = [0,1,2,3,4,5,6,7,8,9]

      ms.sort(collection).should eq sorted_collection

      collection = [7,9,2,3,1,0,4,5,6,8]
      sorted_collection = [0,1,2,3,4,5,6,7,8,9]

      ms.sort(collection).should eq sorted_collection

      collection = [9,8]
      sorted_collection = [8,9]

      ms.sort(collection).should eq sorted_collection

      collection = [0]
      sorted_collection = [0]

      ms.sort(collection).should eq sorted_collection
    end
  end
end
